### PR TITLE
CVE-2016-5145 and CVE-2013-0881

### PR DESCRIPTION
**Pull request by liujustin**

### CODE REVIEW

File: cves/CVE-2013-0881.yml [hunk 1]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Google Chrome allowed remote attackers to cause a denial of service 
+  (incorrect read operation) via crafted data in the Matroska container format.
```
Comment (andymeneely) [inline cves/CVE-2013-0881.yml]: First sentence is pretty much copied from the CVE. I need MUCH more detail here.

File: cves/CVE-2013-0881.yml [hunk 2]
```diff
@@ -116,7 +134,7 @@ interesting_commits:
     emerging themes?
   commits:
   - commit: 
-    note: 
+    note: Not really any interesting commits.
```
Comment (andymeneely) [inline cves/CVE-2013-0881.yml]: You can add an "answer" field above like in major_events. How do I know you did this part? What did you look at? There was a year between the VCC and fix, and nothing interesting happened?

File: cves/CVE-2013-0881.yml [hunk 3]
```diff
@@ -149,37 +167,40 @@ lessons:
...
+    applies: False
     note: 
   yagni:
-    applies: 
-    note: 
+    applies: True
+    note: |
+      In this vulnerability, the person who wrote the code for the audio decoding did not check
+      that the chromium pipeline does not support floats between FFmpegAudioDecoder and AudioRenderer.
+      Because the developer(s) did not forsee this, it caused this vulnerability.
```
Comment (andymeneely) [inline cves/CVE-2013-0881.yml]: I'm not clear on how the first sentence connects to the second here.

File: cves/CVE-2016-5145.yml [hunk 4]
```diff
@@ -120,9 +144,9 @@ interesting_commits:
     emerging themes?
   commits:
   - commit: 
-    note: 
+    note: No interesting commits
```
Comment (andymeneely) [inline cves/CVE-2016-5145.yml]: Same as above - show me that you did this part.

File: cves/CVE-2016-5145.yml [hunk 5]
```diff
@@ -109,8 +131,10 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: |
+    This mistake was found in the code that implemented the HTML5 Structured 
+    Clone algorithm for Chromium's third part webkit.
```
Comment (andymeneely) [inline cves/CVE-2016-5145.yml]: "webkit" is fine

File: cves/CVE-2016-5145.yml [hunk 6]
```diff
@@ -32,7 +32,15 @@ description_instructions: |
...
-description: 
+description: |
+  Blink, a rendering engine used by Chromium, allowed remote attackers to bypass the same-origin policy 
+  restriction for images that was written in the JavaScript code. Same Origin Policy is a concept where 
+  a web browser allows scripts in the first page to access data in the second page but only if they are 
+  from the same origin (URI scheme, host name, and port number). They are able to bypass the policy because 
+  Blink did not ensure the taint property is preserved after a structure-clone operation on an ImageBitmap 
+  object derived from a cross-origin image. What this means is that Blink, after cloning of an ImageBitmap 
+  object, did not keep the property that determined whether it would taint a canvas or not with cross-origin data
+  which would prevent interactivity.
```
Comment (andymeneely) [inline cves/CVE-2016-5145.yml]: Good description.

File: cves/CVE-2013-0881.yml [hunk 7]
```diff
@@ -53,8 +60,17 @@ fixes_vcc_instructions: |
...
 fixes:
 - :commit: b5eca3cffa0afa077d24d98a1f7d9e4b44e5ad4e
-  :note: ''
-vccs: []
+  :note: |
+    Adding AVSampleFormat to SampleFormat converters in FFmpegCommon and decoder support for
+    float planar and float interleaved playback.
+vccs:
+- :commit: 61e4a97449c59ce762603139ee541528b37d5c64
+  :note: Did not check the edge cases for what MP3 can decode to. Chromium does not support S16P.
```
Comment (tcoon) [inline cves/CVE-2013-0881.yml]: Can you elaborate a bit more on how this differs from the other VCC?

File: cves/CVE-2013-0881.yml [hunk 8]
```diff
@@ -105,8 +125,9 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: |
+    This mistake was found in the media subsystem code.
```
Comment (tcoon) [inline cves/CVE-2013-0881.yml]: How did you come to this conclusion? I believe the answer here should be something along the lines of "based on the CVE description and the codebase the files were found in".

File: cves/CVE-2013-0881.yml [hunk 9]
```diff
@@ -53,8 +60,17 @@ fixes_vcc_instructions: |
...
+vccs:
+- :commit: 61e4a97449c59ce762603139ee541528b37d5c64
+  :note: Did not check the edge cases for what MP3 can decode to. Chromium does not support S16P.
+- :commit: 1bae3ad673292afa7915611c3432b313b221c373
+  :note: |
+    Did not check the edge cases for what MP3 can decode to. Chromium does not support S16P.
+    Overlooked possibility that some codecs only output float data. Pipeline does not support 
+    outputting of float data.
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (tcoon) [inline cves/CVE-2013-0881.yml]: +3

File: cves/CVE-2016-5145.yml [hunk 10]
```diff
@@ -55,10 +63,22 @@ fixes_vcc_instructions: |
...
+    Forgot to pass variable to writeImageBitmap function and readImageBitmap boolean function
+    that checks if the origin is clean.
+- :commit: 2128130250de3702dd0e8bcc3c09396e769b3b1f
+  :note: |
+    Forgot to add boolean isImageDataOriginClean that will be used to set whether the image is
+    clean or not when creating a new ImageBitmap.
+- :commit: 455d762d883dae25439ca74d67bd84c297a6e135
+  :note: Did not set whether or not image was clean or not which led to this vulnerability.
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (tcoon) [inline cves/CVE-2016-5145.yml]: +2

File: cves/CVE-2013-0881.yml [hunk 11]
```diff
@@ -115,10 +136,19 @@ interesting_commits:
...
+    note: |
+      This commit, though it was one of the vcc's introduced a lot of new code such as 
+      an audio_splicer and an audio_time_stamp_helper. What is interesting is that the developer,
+      acolwell@chromium.org added very thorough unit tests for both of those files that verified
+      if the code they added worked as intended.
+  - commit: b5eca3cffa0afa077d24d98a1f7d9e4b44e5ad4e
+    note: |
+      This commit is the fix to this vulnerability but there is a something very interesting and peculiar about it.
+      In the file, media/base/audio_decoder_config.cc in the method, AudioDecoderConfig::CopyFrom, dalecurtis@google.com,
+      the developer who fixed this vulnerability removed many of the getters for AudioDecoderConfig from the config.cc file
```
Comment (tcoon) [inline cves/CVE-2013-0881.yml]: Good notes here. Make sure to change "i'm" to "I'm"

File: cves/CVE-2016-5145.yml [hunk 12]
```diff
@@ -153,37 +179,41 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: True
+    note: |
+      Defense in depth can and was applied in this vulnerability. The reason why the vulnerability
```
Comment (tcoon) [inline cves/CVE-2016-5145.yml]: Kind of strange wording for the first sentence. Consider rewriting, or just add "be" after "can"

File: cves/CVE-2016-5145.yml (hunk 6 above)
Comment (tcoon) [reply cves/CVE-2016-5145.yml]: Agreed, this is thorough but not too wordy.

File: cves/CVE-2013-0881.yml (hunk 1 above)
Comment (liujustin) [reply cves/CVE-2013-0881.yml]: fixed

File: cves/CVE-2013-0881.yml (hunk 2 above)
Comment (tydus101) [reply cves/CVE-2013-0881.yml]: Did anyone modify any related code in the file? Were there any other teams working on/around the code during the time? Any commits from outside parties?

**Pull request merged**